plugins/techdocs/src/reader/components/Reader.tsx

```diff
@@ -433,6 +440,12 @@ export const Reader = ({ entityId, onReady }: Props) => {
           <TechDocsNotFound errorMessage={contentErrorMessage} />
         </>
       )}
+
+      {shadowDomRef?.current?.shadowRoot?.innerHTML && (
```

Comment:
Do we want to display the search bar in the docs tab of entity pages? Currently the linking functionality from search redirects to the actual docs page instead of being within entity. Which is a better approach here:
1. Don't make entity page docs tab searchable
2. Keep search also within entity page docs tab. Massage linking to stay within entity page docs
3. Link always to the actual tech docs pages when searching here

Reply:
Kinda leaning towards either 1. or 3. on this mostly just to keep logic at a minimum to try to figure out linking etc. UX would definitely be the best with option number 2. though...

Reply:
The path of least surprise is likely better for the user. Adding inline search to entity page tech docs as well. Introduces some implicit expectations on url patterns for the pages :/

Reply:
There is related discussion around combining the routes in #3122. Wondering if it would be possible to work with both routes depending on the context. Feels like routeRefs should be usable for his sort of thing, but haven't looked into myself.

Reply:
I think based on the thinking in the last comment in #3122, we should really strive for option 2.

I like Andrew's thoughts related to routeRefs, but I don't think there's any functionality that exists today to allow a component in a sub-tree of React to know the context of the route it's rendered in.

Perhaps if we wrapped all routable extensions with a simple context that could return the current active `routeRef`, and then we could use it to return the base off of which we could point result selections?  Like...

```typescript
const activeRouteRef = useActiveRouteRef();
const getBaseDocsUrl = useRouteRef(activeRouteRef);
navigate(`${getBaseDocsUrl()}${document.path}`);
```

Although maybes there's no way to know the documentation sub-route path on an entity?

I dunno, ping @freben @benjdlambert?

Reply:
Hey @Xantier -- based on quick chat w/@Rugvip, an approach like the one I pseudo-coded out would _probably_ work.

There isn't a public API that handles `useActiveRouteRef()` _yet_, but most of the logic and metadata to make it happen exists, and would probably be generally useful (any plugin that has an "inside an entity" + "standalone page" type of usage pattern would likely benefit down the line).

Would you be up for throwing a hook like that together for this?  Similar logic can be found in [RouteResolver](https://github.com/backstage/backstage/blob/master/packages/core-app-api/src/routing/RouteResolver.ts#L105).

Reply:
Thanks for digging deeper into this. I'll have a look and attach implementation of that into this PR.

Reply:
@iamEAP, I finally had some time to look into this and I don't quite see a way forward with using refs with the current implementation that is present in the codebase. Getting an active routeref itself is fine, it is in the end the last matched route that has a ref within it. 

The problem lies with the refs themselves. Catalog for example has entitytriplet attached to it's route but `docs` for example is a no-param route and therefore couldn't be constructed with the pseudocode above. 

Maybe I am looking at this completely the wrong way...

Reply:
Ah, I think what's missing from the `docs` side is [this](https://github.com/backstage/backstage/pull/6458/files#diff-c6453d0dd45f236e992074aae7ead2217f5fc579a4a3420f2ed66af9a02c7df2R60) upcoming change?

Reply:
Yup, that would likely do it I believe. We can take few ways forward here I guess if we want to advance with this PR:
1. Merge as is, we would support linking to both docs pages and entitydocs tabs with the caveat that routes are hardcodish. Then later implement routeref based discovery of linking
2. I'll push the current naive implementation of `useActiveRouteRef` which uses the last routeRef in the matched array of refs from the context. We'll still use different linking for docs pages and entitydocs tabs. The we'll later reimplement the linking again. 

I'd be proponent of the first option since I'm not 100% if the naive implementation will work without modifications when the above change is added in.

Reply:
Heyo!  Yeah, maybe option 1 is fine for now and we can create a follow-up?  Or, perhaps to avoid adding something to the `<Reader />` API and then immediately deprecating it, we could _only_ support one or the other for now, with a follow-up issue to pull the ref dynamically from context?  What do y'all think?

@emmaindal